relentless/variable.py

```diff
@@ -559,6 +702,144 @@ class BinaryOperator(DependentVariable):
+
+    Parameters
+    ----------
+    a : :class:`Variable`
+        First value.
+    b : :class:`Variable`
+        Second value.
+
+    """
+    @property
+    def value(self):
+        return self.a.value+self.b.value
+
+    def _derivative(self, param):
+        if param == 'a':
+            return 1.0
+        elif param == 'b':
+            return 1.0
+        else:
+            raise ValueError('Unknown parameter')
+
+class Difference(BinaryOperator):
+    r"""Difference of two values.
+
+    Given two values :math:`a` and :math:`b`, this returns :math:`a-b`
+
+    Parameters
+    ----------
+    a : :class:`Variable`
+        First value.
+    b : :class:`Variable`
+        Second value.
+
+    """
+    @property
+    def value(self):
+        return self.a.value-self.b.value
+
+    def _derivative(self, param):
+        if param == 'a':
+            return 1.0
+        elif param == 'b':
+            return -1.0
+        else:
+            raise ValueError('Unknown parameter')
+
+class Product(BinaryOperator):
+    r"""Product of two values.
+
+    Given two values :math:`a` and :math:`b`, this returns :math:`ab`
+
+    Parameters
+    ----------
+    a : :class:`Variable`
+        First value.
+    b : :class:`Variable`
+        Second value.
+
+    """
+    @property
+    def value(self):
+        return self.a.value*self.b.value
+
+    def _derivative(self, param):
+        if param == 'a':
+            return self.b.value
+        elif param == 'b':
+            return self.a.value
+        else:
+            raise ValueError('Unknown parameter')
+
+class Quotient(BinaryOperator):
+    r"""Quotient of two values.
+
+    Given two values :math:`a` and :math:`b`, this returns :math:`\frac{a}{b}`
+
+    Parameters
+    ----------
+    a : :class:`Variable`
+        First value.
+    b : :class:`Variable`
+        Second value.
+
+    """
+    @property
+    def value(self):
+        try:
+            return self.a.value/self.b.value
+        except ZeroDivisionError:
+            return numpy.inf
+
+    def _derivative(self, param):
+        if param == 'a':
+            try:
+                return 1.0/self.b.value
+            except ZeroDivisionError:
+                return numpy.inf
+        elif param == 'b':
+            try:
+                return -self.a.value/self.b.value**2.0
+            except ZeroDivisionError:
+                return numpy.inf
+        else:
+            raise ValueError('Unknown parameter')
+
+class Power(BinaryOperator):
+    r"""Takes a value to a power.
+
+    Given two values :math:`a` and :math:`b`, this returns :math:`a^b`
+
+    Parameters
+    ----------
+    a : :class:`Variable`
+        First value.
+    b : :class:`Variable`
+        Second value.
+
+    """
+    @property
+    def value(self):
+        return self.a.value**self.b.value
+
+    def _derivative(self, param):
+        if param == 'a':
+            return self.b.value*self.a.value**(self.b.value-1.0)
```

Comment:
```suggestion
            return self.b.value*self.a.value**(self.b.value-1)
```
There should also be a special case for `b == 0`, right?